src/System.IO.FileSystem.Watcher/ref/System.IO.FileSystem.Watcher.cs

```diff
@@ -81,10 +81,10 @@ public partial struct WaitForChangedResult
     {
         private object _dummy;
         private int _dummyPrimitive;
-        public System.IO.WatcherChangeTypes ChangeType { get { throw null; } set { } }
-        public string Name { get { throw null; } set { } }
-        public string OldName { get { throw null; } set { } }
-        public bool TimedOut { get { throw null; } set { } }
+        public System.IO.WatcherChangeTypes ChangeType { readonly get { throw null; } set { } }
+        public string Name { readonly get { throw null; } set { } }
+        public string OldName { readonly get { throw null; } set { } }
+        public bool TimedOut { readonly get { throw null; } set { } }
```

Comment:
This is interesting.  The C# compiler is adding `readonly` automatically to auto-prop get accessors.  It's a breaking change to remove `readonly` from a property once it's there, which means if we ever wanted to make the getter mutating, we'd have issues.  I don't see that being a problem in this case, but it's something to think about.

Reply:
i.e. moving forward any time we see an auto-prop get accessor replaced by an actual implementation, we need to take a hard look at it.  Presumably at that point our compat tooling will tell us that something that used to be readonly no longer is in the implementation?

Reply:
It also technically means that when someone changes a public property's implementation in corefx, e.g.
```C#
private int _foo;
public int Foo { get => _foo; set => _foo = value; }
```
to instead be:
```C#
public int Foo { get; set; }
```
they're actually changing the public surface area, the ref will need to be updated, etc.

cc: @terrajobst

Reply:
We should catch these changes now.  
I think missing these from the refs is a bug in GenAPI where it's not doing the right thing to generate the readonly keyword all the time.  GenAPI should be able to see the readonly attribute that the compiler added and emit readonly.  I haven't yet root caused that /cc @tannergooding.

Reply:
CC. @jaredpar, I believe the compiler team stated its a behavioral change not necessarily a breaking change.

Reply:
I double checked all the places I had to add `readonly` and can confirm that there is no GenAPI bug, so we should be in a good spot with the added validation.  When it flags things they will be real problems, and running GenAPI can fix them.